### DISCUSSION TITLE
Why `stimulus-vite-helpers`?

### DISCUSSION THREAD

**Discussion by mabras** [ElMassimo/stimulus-vite-helpers#7] [Q&A] [answered]
What is the difference between `stimulus-vite-helpers` and simply just using:

```
# app/frontend/entrypoints/application.js

import "~/controllers/index.js";
```

```
# app/frontend/controllers/index.js

import { application } from "./application"

import SomeController from "./some_controller_controller.js"
application.register("some", SomeController)

...
```

```
# app/frontend/controllers/application.js

import { Application } from "@hotwired/stimulus"

const application = Application.start()

// Configure Stimulus development experience
application.debug = false
window.Stimulus   = application

export { application }
```

**Comment by ElMassimo** [comment, marked as answer]
Hi Mohamad!

There's no difference, they are both equivalent at build time.

This package is intended to make it easier to [migrate](https://stimulus.hotwired.dev/handbook/installing#using-webpack-helpers) applications that have a large amount of controllers, and to allow skipping the boilerplate of manually registering controllers, using the filenames as the naming convention as is typical of Rails.

If you are using Vite, another advantage is that it ensures your controllers work out of the box with [`vite-plugin-stimulus-hmr`](https://github.com/ElMassimo/vite-plugin-stimulus-hmr).
